src/cargo/core/resolver/mod.rs

```diff
@@ -1091,12 +1088,9 @@ fn activate_deps_loop(
                             .filter_map(|(_, (deb, _, _))| {
```

Comment:
I think the `filter_map` + `next` combo can be replaced with `find`

Reply:
something like?
```
.filter_map(|(_, (new_dep, _, _))| past_conflicting_activations.get(&new_dep))
.flat_map(|x| x)
.find(|con| cx.is_conflicting(None, con))
```
I'd love to find the elegant way to do this, just haven't found it yet.

Reply:
Oh right yeah, don't worry about this it's fine as-is!

Reply:
Submitted a PR to the std to make this code prettier :) https://github.com/rust-lang/rust/pull/49098